### PR TITLE
Handle a 1D array from 2D array in legacy-array-sections

### PR DESCRIPTION
**Pull request by HarshitaKalani**
Fixes: https://github.com/lfortran/lfortran/issues/2481

### CODE REVIEW

File: integration_tests/legacy_array_sections_02.f90 [hunk 1]
```diff
@@ -0,0 +1,11 @@
+subroutine sub(a)
+    double precision :: a
+    dimension :: a(16)
+end subroutine
+    
+program main
+    double precision :: a(16,3)
+    integer :: j
```
Comment (certik) [inline integration_tests/legacy_array_sections_02.f90:8]: Let's add here:
```
j = 3
a = 1
a(16,3) = 2
```

File: integration_tests/legacy_array_sections_02.f90 [hunk 2]
```diff
@@ -0,0 +1,11 @@
+subroutine sub(a)
+    double precision :: a
+    dimension :: a(16)
```
Comment (certik) [inline integration_tests/legacy_array_sections_02.f90:3]: Let's add here:
```
if (abs(a(1) - 2) > 1d-15) error stop
if (.not. all(abs(a(2:) - 1) < 1d-15)) error stop
```

**Review by certik — COMMENTED** [review]
Excellent, thanks for the fix. Since we are working on this, let's make it more robust. I left some comments how to do so. Test them locally first, they might not work (in that case we'll use the current approach).

File: integration_tests/legacy_array_sections_02.f90 (hunk 2 above)
Comment (HarshitaKalani) [reply integration_tests/legacy_array_sections_02.f90:3]: Here it should be
```diff
- if (abs(a(1) - 2) > 1d-15) error stop
+ if (abs(a(1) - 1) > 1d-15) error stop
```
right? Since a(1) = 1.

Comment (certik) [reply integration_tests/legacy_array_sections_02.f90:3]: It should be 2, due to `    a(16,3) = 2`.

File: src/lfortran/semantics/ast_body_visitor.cpp [hunk 3]
```diff
@@ -2528,8 +2528,16 @@ class BodyVisitor : public CommonVisitor<BodyVisitor> {
...
                                 ASR::asr_t* descriptor_array = ASR::make_Array_t(al, x.base.base.loc, ASRUtils::type_get_past_array(expected_arg_type),
                                                                 dims.p, dims.size(), ASR::array_physical_typeType::DescriptorArray);
+                                Vec<ASR::dimension_t> dims2;
+                                dims2.reserve(al, 1);
+                                ASR::dimension_t dim2;
+                                dim2.loc = x.base.base.loc;
+                                ASR::Array_t* array_t = ASR::down_cast<ASR::Array_t>(expected_arg_type);
+                                dim2.m_length = array_t->m_dims[0].m_length;
+                                dim2.m_start = nullptr;
+                                dims2.push_back(al, dim2);
```
Comment (certik) [inline src/lfortran/semantics/ast_body_visitor.cpp:2538]: This is not needed anymore I think.

**Comment by certik** [comment]
I think the test now doesn't run in gfortran --- make sure it works for you locally.

**Comment by certik** [comment]
I can investigate later today. The PR looks good. All I want is to design the test in such a way as to test that the array was shifted/sliced correctly.

**Review by certik — APPROVED** [review]
I updated the test to check for correctness. I also removed extra trailing white space. 

I think it looks great now, thanks!

**Pull request merged**